File: fibo.c
#include <math.h>
#include <stdio.h>

int Fibonacci(int n) {
  const double phi = (1 + sqrt(5)) / 2;
  const double psi = (1 - sqrt(5)) / 2;

  // Use Binet's formula
  double fib_n = (pow(phi, n) - pow(psi, n)) / sqrt(5);

  // Round the result to nearest integer
  return (int)(fib_n + 0.5);
}

int main() {
  int n = 40;
  printf("Fibonacci(%d) = %d\n", n, Fibonacci(n));
  return 0;
}

File: fibo.py
import math

def fibonacci(n):
    phi = (1 + math.sqrt(5)) / 2
    psi = (1 - math.sqrt(5)) / 2
    
    fib_n = (phi**n - psi**n) / math.sqrt(5)
    return round(fib_n)

if __name__ == "__main__":
    n = 400
    print(f"Fibonacci({n}) = {fibonacci(n)}")

